backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRootBranch.java

```diff
@@ -61,4 +69,10 @@ public GitMacheteRootBranch(String name, List<GitMacheteNonRootBranch> downstrea
   public Option<String> getCustomAnnotation() {
     return Option.of(customAnnotation);
   }
+
+  @Override
+  @Pure
```

Comment:
Again, not `@Pure` :/ would be pure if we considered `equals` equality... but not in case of reference equality (which makes more sense for static analysis)

Reply:
Leave the anno but add comment